dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/acl/TrackerProgramServiceTest.java

```diff
@@ -0,0 +1,124 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.acl;
+
+import static org.hisp.dhis.common.IdentifiableObjectUtils.getUids;
+import static org.hisp.dhis.test.utils.Assertions.assertIsEmpty;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.io.IOException;
+import java.util.List;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.security.acl.AccessStringHelper;
+import org.hisp.dhis.test.integration.PostgresIntegrationTestBase;
+import org.hisp.dhis.trackedentity.TrackedEntityType;
+import org.hisp.dhis.tracker.TestSetup;
+import org.hisp.dhis.user.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class TrackerProgramServiceTest extends PostgresIntegrationTestBase {
+  @Autowired private TestSetup testSetup;
+  @Autowired private TrackerProgramService trackerProgramService;
+  @Autowired private IdentifiableObjectManager manager;
+
+  private User regularUser;
+
+  @BeforeAll
+  void setUp() throws IOException {
+    testSetup.importMetadata();
+    regularUser = userService.getUser("Z7870757a75");
+    injectSecurityContextUser(regularUser);
+  }
+
+  @Test
+  void shouldReturnAllAccessiblePrograms() {
+    List<String> accessiblePrograms =
+        List.of(
+            "BFcipDERJnf",
+            "shPjYNifvMK",
+            "pcxIanBWlSY",
+            "UWRnoyBjvqi",
+            "YlUmbgnKWkd",
+            "SeeUNWLQmZk",
+            "sLngICFQjvH",
+            "TsngICFQjvH",
+            "TsngICFQjvP");
+
+    assertEquals(accessiblePrograms, getUids(trackerProgramService.getAccessibleTrackerPrograms()));
+  }
+
+  @Test
+  void shouldReturnAllAccessibleProgramsByTrackedEntityType() {
+    List<String> accessiblePrograms =
+        List.of(
+            "BFcipDERJnf",
+            "shPjYNifvMK",
+            "pcxIanBWlSY",
+            "UWRnoyBjvqi",
+            "YlUmbgnKWkd",
+            "SeeUNWLQmZk",
+            "sLngICFQjvH");
+
+    TrackedEntityType trackedEntityType = manager.get(TrackedEntityType.class, "ja8NY4PW7Xm");
+    assertNotNull(trackedEntityType);
+    assertEquals(
+        accessiblePrograms,
+        getUids(trackerProgramService.getAccessibleTrackerPrograms(trackedEntityType)));
+  }
+
+  @Test
+  void shouldReturnEmptyListWhenNoProgramAccessible() {
+    makeProgramInaccessible("TsngICFQjvP");
```

Comment:
I would expect us to have two tests making the same call, using the same input. The result of one test should then include a program and the other should not.

It looks like `shouldReturnAllAccessibleProgramsByTrackedEntityType` and this test `shouldReturnEmptyListWhenNoProgramAccessible` are those tests but the use a different TET as input.